### ISSUE TITLE
No axis projection or oblique projection split criterion exist as regression criteria

### ISSUE THREAD

**Issue by morgsmss7** [NeuroDataDesign/scikit-learn#4]
#### Description
There are two split criteria that are not included as regression criteria in sklearn: Axis projection in which one predictor is chosen at random to calculate MSE and Oblique projection in which outputs are projected onto a sparse line before splitting with MSE. More tests are being done to show when they perform differently than the existing criteria.

#### Enhancement
I plan to implement these two split criteria as subclasses to RegressionCriteria
